datafusion/physical-plan/src/sorts/stream.rs

```diff
@@ -48,6 +48,10 @@ pub trait PartitionedStream: std::fmt::Debug + Send {
     ) -> Poll<Option<Self::Output>>;
 }
 
+/// A fallible [`PartitionedStream`] of [`Cursor`](super::cursor::Cursor) and [`RecordBatch`]
+pub(crate) type CursorStream<C> =
+    Box<dyn PartitionedStream<Output = Result<(C, RecordBatch)>>>;
```

Comment:
Is the idea this will eventually just be `Box<dyn PartitionedStream<Output = Result<C>>>` with the `RecordBatch` handled separately?